libs/hdf-converters/src/converters-from-hdf/caat/reverse-caat-mapper.ts

```diff
@@ -0,0 +1,257 @@
+    const controlSeverity =
+      hdf.severity === 'medium' ? 'moderate' : hdf.severity;
+    return this.fix(hdf.wraps.impact === 0 ? 'none' : controlSeverity);
+  }
+
+  getRow(control: ContextualizedControl, filename: string): CAATRow[] {
+    const hdf = control.hdf;
+    const allRows: CAATRow[] = _.compact(
+      hdf
+        .canonized_nist(FromHDFToCAATMapper.NistCanonizationConfig)
+        .map((formattedNistTag) => {
+          // I don't think the canonized_nist function would return something that wasn't a nist tag, but this check was there originally quite probably for a reason, but I've not been able to find a sample that triggers this case myself
+          if (!formattedNistTag) {
+            // early exiting forces us to use the compact function to get rid of empty values in the array
+            return;
+          }
+
+          const row: CAATRow = {};
+          row['Control Number'] = formattedNistTag;
+          row['Finding Title'] = `Test ${this.fix(hdf.wraps.id)} - ${this.fix(
+            hdf.wraps.title
+          )}`;
+          if (hdf.start_time) {
+            row['Date Identified'] = FromHDFToCAATMapper.formatDate(
+              new Date(hdf.start_time),
+              '/'
+            );
+          }
+          row['Finding ID'] = `${filename} - Test ${this.fix(hdf.wraps.id)}`;
+          row['Finding Description'] = this.fix(hdf.wraps.title);
+          row['Weakness Description'] = this.createCaveat(hdf);
+          row['Control Weakness Type'] = 'Security';
+          row['Source'] = 'Self-Assessment';
+          row['Test Method'] = 'Test';
+          row['Test Objective'] = this.fix(
+            hdf.descriptions.check ?? hdf.wraps.tags.check
+          );
+          row['Test Result Description'] = this.fix(
+            this.createTestResultDescription(hdf)
+          );
+          row['Test Result'] = this.createTestResult(hdf);
+          row['Recommended Corrective Action(s)'] = this.fix(
+            hdf.descriptions.fix ?? hdf.wraps.tags.fix
+          );
+          row['Impact'] = this.createImpact(hdf);
+          return row;
+        })
+    );
+    return allRows;
+  }
+
+  // returnWorkBook: true -> raw workbook class
+  // returnWorkBook: false | undefined -> binary string
+  toCAAT(returnWorkBook?: false) {
+    // Sheet names must be unique across the workbook
+    const takenSheetNames: string[] = [];
+
+    // Define our workbook
+    const wb = XLSX.utils.book_new();
+
+    // For each contextualized evaluation
+    for (const d of this.data) {
+      // Ensure sheet name uniqueness
+      let renameCount = 2;
+      let sheetName: string = `${
+        d.filename ?? d.data.data.profiles.at(0)?.name ?? 'ExecJSON'
+      }`.substring(0, 31);
+      while (takenSheetNames.includes(sheetName)) {
+        sheetName =
+          `${
+            d.filename ?? d.data.data.profiles.at(0)?.name ?? 'ExecJSON'
+          } `.substring(0, 26) + ` (${renameCount})`;
```

Comment:
Why is this getting the substring(0, 25)? Is that the number of digits allowed for a sheet name?

Reply:
sheet names can be at max 31 characters.  i should probs make that a constant.  the 26 is more or less arbitrary so as to have enough space to put in the rename count